packages/block-library/src/home-link/index.php

```diff
@@ -151,9 +151,20 @@ function render_block_core_home_link( $attributes, $content, $block ) {
 		$aria_current = ' aria-current="page"';
 	}
 
+	// If the home link is inside a navigation block, it should be wrapped in a list item with additional attributes.
+	// Use the block context to determine if the home link is inside a navigation block.
+	$in_navigation = ! empty( $block->context );
+
+	if ( $in_navigation ) {
+		$item_markup = '<li %1$s><a class="wp-block-home-link__content wp-block-navigation-item__content" href="%3$s" rel="home"%4$s>%5$s</a></li>';
+	} else {
+		$item_markup = '<a %2$s href="%3$s" rel="home"%4$s>%5$s</a>';
+	}
```

Comment:
I think this may be better covered via the `block_core_navigation_listable_blocks` filter introduced in 6.5.

Or you can add the block to the default array here: https://github.com/WordPress/gutenberg/blob/e4623e1adfefa46eb001ea42a87ada457d52be1b/packages/block-library/src/navigation/index.php#L23-L26

Reply:
Interesting. Thanks I will look those up!

Reply:
Though how would I avoid it breaking on 6.3, 6.4?

Reply:
I'm mostly thinking out loud, but this block would also need the current-menu-item class on the li.